capstone/src/main/java/com/google/sps/data/ApkUnzipContent.java

```diff
@@ -0,0 +1,87 @@
+package com.google.sps.data;
+
+import com.google.appengine.api.datastore.Entity;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+
+public class ApkUnzipContent {
+  /* Creates a class for each APK uploaded before unzipping for easy structuring of APK contents and testing */
+
+  // Declare variable for storage
+  public long timestamp = System.currentTimeMillis();
+  
+  // Declare map to store individual APK files and their characteristics
+  private HashMap<String, ArrayList<Long>> apkFilesListMap = new HashMap<String, ArrayList<Long>>();
+  
+  // Declare map to store categorized APK contents according to file types
+  private HashMap<String, ArrayList<Long>> apkPackageContentMap = new HashMap<String, ArrayList<Long>>();
+  
+  public HashMap<String,ArrayList<Long>> getApkPackageContentMap() {
+    return this.apkPackageContentMap;
+  }
+
+  public HashMap<String,ArrayList<Long>> getApkFilesListMap() {
+    return this.apkFilesListMap;
+  }
+
+  // Creates a constructor class for each APK with initialized map values to avoid null exceptions that may arise in Datastore from the absence of any category in APK
+  public ApkUnzipContent() {
+    apkPackageContentMap.put("res", new ArrayList<Long>(Arrays.asList(0L, 0L)));
+    apkPackageContentMap.put("dex", new ArrayList<Long>(Arrays.asList(0L, 0L)));
+    apkPackageContentMap.put("assets", new ArrayList<Long>(Arrays.asList(0L, 0L)));
+    apkPackageContentMap.put("lib", new ArrayList<Long>(Arrays.asList(0L, 0L)));
+    apkPackageContentMap.put("arsc", new ArrayList<Long>(Arrays.asList(0L, 0L)));
+    apkPackageContentMap.put("misc", new ArrayList<Long>(Arrays.asList(0L, 0L)));
+
+  }
```

Comment:
Good thinking! Using hashmaps will give O(1) lookups but there's a small problem. The chosen categories have sub directories which have numerous files in them. If we are allowing users to view their APKs, I feel it's a bit vague to just show them the top level of the APK. It's just like saying, "Hey your res directory is too huge" then the user responds, "Okay, but what file in the res directory is accounting for that huge size?" Also separate chaining might lead to performance hits in the long run, because most of these folders have numerous files the user could explore. I suggest you take a critical look at it but then again it's just a suggestion so you could take it with a grain of salt.

Reply:
I think that's a good suggestion. That feels like a followup implementation. The data per file is also being calculated but is not yet being displayed.